lib/ohai/plugins/packages.rb

```diff
@@ -50,7 +50,16 @@
 
       pkgs.each do |pkg|
         name, epoch, version, release, installdate, arch = pkg.split
-        packages[name] = { "epoch" => epoch, "version" => version, "release" => release, "installdate" => installdate, "arch" => arch }
+        if packages[name]
+          # We have more than one package with this exact name!
+          # Create an "other_versions" array for tracking all versions of packages with this name.
+          # The top-level package information will be the first one returned by rpm -qa,
+          # all others go in this list, with the same information they'd normally have.
+          packages[name]["other_versions"] = [] unless packages[name]["other_versions"]
+          packages[name]["other_versions"] << Mash.new({ "epoch" => epoch, "version" => version, "release" => release, "installdate" => installdate, "arch" => arch })
```

Comment:
Any chance any of these would ever be unique so we could just use them as a key for a hash?

Reply:
Not unless you want to concatenate stuff. By definition you need name+version+release+architecture to guarantee uniqueness. You could roll the dice with epoch or installdate, but those aren't _guaranteed_ unique (and I suspect there are edge cases where they break, like install discs or similar). An array seems like the least bad option.

Reply:
the NEVRA itself (name-epoch-version-release-arch) should be unique